clients/java/src/main/java/io/camunda/zeebe/client/impl/http/TypedApiEntityConsumer.java

```diff
@@ -91,13 +93,23 @@ public JsonApiEntityConsumer(
 
     @Override
     public ApiEntity<T> generateContent() throws IOException {
-      buffer.asParserOnFirstToken();
-
-      if (isResponse) {
-        return ApiEntity.of(json.readValue(buffer.asParserOnFirstToken(), type));
+      try {
+        if (isResponse) {
+          return ApiEntity.of(json.readValue(buffer.asParserOnFirstToken(), type));
+        }
+        return ApiEntity.of(json.readValue(buffer.asParserOnFirstToken(), ProblemDetail.class));
+      } catch (final IOException ioe) {
+        // write the original JSON response into an error response
+        final ByteArrayOutputStream output = new ByteArrayOutputStream();
+        final JsonGenerator generator = json.createGenerator(output);
+        buffer.serialize(generator);
```

Comment:
`serialize` itself throws an `IOException`, I guess we handle it upstreams as before?

Reply:
We'll handle it explicitly to avoid obfuscating the original issue. However, I don't see much we can do anymore except for saying "we received something we cannot decipher"